hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/service/SnapshotDeletingService.java

```diff
@@ -172,9 +174,21 @@ public BackgroundTaskResult call() throws Exception {
                 " is not found", BUCKET_NOT_FOUND);
```

Comment:
We can throw `IllegalStateException` instead since `bucketInfo` should not be null under normal circumstances.

And at least make the error message very very clear that OM is not in an expected state.